platform/core.multitabs/src/org/netbeans/core/multitabs/impl/TabTableUI.java

```diff
@@ -173,6 +193,28 @@ public void mouseDragged( MouseEvent e ) {
             @Override
             public void mouseMoved( MouseEvent e ) {
                 orig.mouseMoved( e );
+
+                if (hoverSupport) {
+                    Point p = e.getPoint();
+                    int row = table.rowAtPoint( p );
+                    int column = table.columnAtPoint( p );
+                    setHover(row, column);
+                }
+            }
+
+            private void setHover( int row, int column ) {
+                if (row == hoverRow && column == hoverColumn) {
+                    return;
+                }
+
+                if (hoverRow != -1 && hoverColumn != -1) {
+                    table.repaint(table.getCellRect(hoverRow, hoverColumn, true));
+                }
+                if (row != -1 && column != -1) {
+                    table.repaint(table.getCellRect(row, column, true));
+                }
+                hoverRow = row;
+                hoverColumn = column;
```

Comment:
Don't you need to set these _before_ calling repaint? The repaint typically ends up being scheduled for later, but it's cleaner to not assume this.

Reply:
This does not work. The first repaint is necessary to repaint the tab that was hovered before (remove hover background). The second repaint repaints the currently hovered tab. Maybe I should add some comments...

Reply:
I meant like this:

```
int oldRow = hoverRow;
int oldColumn = hoverColumn;

if (oldRow != -1 && oldColumn != -1) {
    table.repaint(table.getCellRect(oldRow, oldColumn, true));
}
if (row != -1 && column != -1) {
    table.repaint(table.getCellRect(row, column, true));
}
hoverRow = row;
hoverColumn = column;
```
Since regardless of whether you are updating the new area or the old area, you want paintComponent to run according to the _new_ state.

Reply:
(Note: I edited this comment--see the latest version.)

Reply:
Ok, now I understand 👍 
You're right.
I'll change it.

Reply:
Done